### PR TITLE
Dashboard v2 files page

### PR DESCRIPTION
**Pull request by meeh0w**
# `/files` page

> As always, reviewing commit-by-commit will be easier to follow through

## Big-picture change
* Adds `/files` page. Just visual stuff and basic data fetching.
* Filtering, sorting and all the context-menu stuff is not functional yet. You'll notice a bunch of `console.log()`s as callbacks -- they'll be replaced with proper stuff once I get to implementing the above.

### Breakdown of smaller changes
* adds new icons: `Share`, `Search` and `Copy`
* re-implements `TextInputIcon` component to allow it to be more flexible
* fixes behavior out `defaultTab` prop on `Tabs` component
* styling fixes for table headers
* migrates `Switch` component to `styled-components` approach
* adds a new component called `PopoverMenu` (to serve as a context menu)
* adds new component called `CopyButton`

## Visual overview

https://user-images.githubusercontent.com/3853033/158056550-872f0188-012f-4bb8-a435-44ccb738c4cf.mov

### CODE REVIEW

**Review by peterjan — APPROVED** [review]
perfect 👍

File: packages/dashboard-v2/src/components/FileList/useFormattedFilesData.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useMemo } from "react";
+import prettyBytes from "pretty-bytes";
```
Comment (kwypchlo) [inline packages/dashboard-v2/src/components/FileList/useFormattedFilesData.js:2]: Just a note of something I noticed, I usually try to name my imports either as official docs say or just camel case the name of the library like you did here so it's consistent across the codebase. I noticed in some past pull requests you used different names for the import of pretty-bytes and while it's not a problem or an issue, I like to keep the names of the imports consistent across the codebase and the above mentioned guidelines help me with that.

Screenshot 2022-03-16 at 10 36 03

Feel free to resolve this comment once read, it's just a note.

File: packages/dashboard-v2/src/components/FileList/FileList.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { Switch } from "../Switch";
+import { TextInputIcon } from "../TextInputIcon/TextInputIcon";
+import { SearchIcon } from "../Icons";
+
+import FileTable from "./FileTable";
+import useFormattedFilesData from "./useFormattedFilesData";
+
+const FileList = ({ type }) => {
+  const isMediumScreenOrLarger = useMedia(`(min-width: ${theme.screens.md})`);
+  const { data, error } = useSWR(`user/${type}?pageSize=10`);
```
Comment (kwypchlo) [inline packages/dashboard-v2/src/components/FileList/FileList.js:18]: I think it should be prefixed with `/` to make sure we're not appending this to some subpage

`/user/${type}?pageSize=10`

also not sure if this worked but all api calls **I think** are prefixed with `/api` so `/api/user/${type}?pageSize=10` (not sure if this comment is not relevant in some other useSWR usages too)

Screen from account.siasky.net network tab:
Screenshot 2022-03-16 at 10 48 19

**Review by kwypchlo — CHANGES_REQUESTED** [review]

File: packages/dashboard-v2/src/components/FileList/FileList.js (hunk 2 above)
Comment (meeh0w) [reply packages/dashboard-v2/src/components/FileList/FileList.js:18]: Yeah, the `/api` part is added in the SWR config as a `baseUrl` (line 5), same as prefixing the key with slash (line 8):

https://github.com/SkynetLabs/skynet-webportal/blob/3128a2e7805c1d847e2591b8178fbf2acc0844fc/packages/dashboard-v2/src/lib/swrConfig.js#L4-L8

File: packages/dashboard-v2/src/components/FileList/useFormattedFilesData.js (hunk 1 above)
Comment (meeh0w) [reply packages/dashboard-v2/src/components/FileList/useFormattedFilesData.js:2]: Good point, I'll make it consistent in the upcoming PR 👍

**Review by kwypchlo — APPROVED** [review]

**Pull request merged**